pkgs/applications/networking/browsers/palemoon/default.nix

```diff
@@ -112,24 +170,42 @@ in stdenv.mkDerivation rec {
     mk_add_options MOZ_MAKE_FLAGS="-j${if enableParallelBuilding then "$NIX_BUILD_CORES" else "1"}"
     mk_add_options AUTOCONF=${autoconf213}/bin/autoconf
     '
+
+    runHook postConfigure
   '';
 
-  buildPhase = "./mach build";
+  buildPhase = ''
+    runHook preBuild
+
+    ./mach build
+
+    runHook postBuild
+  '';
 
   installPhase = ''
+    runHook preInstall
+
     ./mach install
 
     # Fix missing icon due to wrong WMClass
+    # TODO report upstream
     substituteInPlace ./palemoon/branding/official/palemoon.desktop \
       --replace 'StartupWMClass="pale moon"' 'StartupWMClass=Pale moon'
     desktop-file-install --dir=$out/share/applications \
       ./palemoon/branding/official/palemoon.desktop
 
+    # Install official branding icons
     for iconname in default{16,22,24,32,48,256} mozicon128; do
       n=''${iconname//[^0-9]/}
       size=$n"x"$n
       install -Dm644 ./palemoon/branding/official/$iconname.png $out/share/icons/hicolor/$size/apps/palemoon.png
     done
+
+    # Remove unneeded SDK data from installation
+    # TODO: move to a separate output?
+    rm -rf $out/{include,share/idl,lib/palemoon-devel-${version}}
```

Comment:
Are they used for something interesting, like creating plugins or something like?

Reply:
I really don't know. Our Firefox package has done the same thing for the last 7 years, so I assume it doesn't matter?

https://github.com/NixOS/nixpkgs/commit/5c5f11560329f41573c862b2146f6f7bc5e19d1b#diff-4f162c39db036b9d3f8839850efcaad3ca502055d04f759ea76555607aa151e0R92-R94

Reply:
Firefox deprecated/invalidated xulrunner, whereas Palemoon was explicitly created/forked because of that Firefox decision (among many others).

If those deleted files are used as a SDK/API for palemoon plugins, creating a `palemoon-sdk` package would be better.

Reply:
Oh… We should prolly just ask on the forum then.

Why an extra package instead of a `dev`/`lib` output?

Reply:
> Why an extra package instead of a `dev`/`lib` output?

This is what I meant by "extra package". Sorry by my confusion.

Reply:
> Oh… We should prolly just ask on the forum then.

https://forum.palemoon.org/viewtopic.php?f=37&t=26796&p=214676#p214675

Upstream says we're installing it incorrectly, we should use `mach package` and extract the resulting archive where we need it instead of `mach install`ing everything. It makes a difference, they assured me. I'll push a WIP PR in abit where we should try to fix that.

Reply:
Spoke too soon, the devs are disagreeing with each other now… I'll wait until they've decided on something. :sweat:

Reply:
They are very temperamental indeed!

Reply:
> Well I just woke up so mah brain may be fetching obsolete data. If so, sorry about that.

Let's leave the past in the past. Negative things stays in your head longer than positive ones. They apologised and that's that, I wouldn't take it personal. ¯\\\_(ツ)\_/¯

They decided that `install` is preferred & deleting the SDK cruft to match what `package` would produce is fine.

Reply:
Less bad! Then, everything is OK after all.